x-pack/plugins/fleet/public/applications/integrations/sections/epm/screens/detail/assets/assets.tsx

```diff
@@ -0,0 +1,252 @@
+    case KibanaAssetType.search:
+      return `/discover/${id}`;
+    case KibanaAssetType.visualization:
+      return `/visualize/edit/${id}`;
+    default:
+      return undefined;
+  }
+};
+
+export const AssetsPage = ({ packageInfo }: AssetsPanelProps) => {
+  const { name, version } = packageInfo;
+  const {
+    savedObjects: { client: savedObjectsClient },
+    http,
+  } = useStartServices();
+
+  const { getPath } = useLink();
+  const getPackageInstallStatus = useGetPackageInstallStatus();
+  const packageInstallStatus = getPackageInstallStatus(packageInfo.name);
+
+  const [assetSavedObjects, setAssetsSavedObjects] = useState<undefined | AssetSavedObject[]>();
+  const [fetchError, setFetchError] = useState<undefined | Error>();
+  const [isLoading, setIsLoading] = useState<boolean>(true);
+
+  useEffect(() => {
+    const fetchAssetSavedObjects = async () => {
+      if ('savedObject' in packageInfo) {
+        const {
+          savedObject: { attributes: packageAttributes },
+        } = packageInfo;
+
+        if (
+          !packageAttributes.installed_kibana ||
+          packageAttributes.installed_kibana.length === 0
+        ) {
+          setIsLoading(false);
+          return;
+        }
+
+        try {
+          const { savedObjects } = await savedObjectsClient.bulkGet(
+            packageAttributes.installed_kibana.map(({ id, type }) => ({
+              id,
+              type,
+            }))
+          );
+          setAssetsSavedObjects(savedObjects as AssetSavedObject[]);
+        } catch (e) {
+          setFetchError(e);
+        } finally {
+          setIsLoading(false);
+        }
+      } else {
+        setIsLoading(false);
+      }
+    };
+    fetchAssetSavedObjects();
+  }, [savedObjectsClient, packageInfo]);
+
+  // if they arrive at this page and the package is not installed, send them to overview
+  // this happens if they arrive with a direct url or they uninstall while on this tab
+  if (packageInstallStatus.status !== InstallStatus.installed) {
+    return (
+      <Redirect
+        to={getPath('integration_details_overview', {
+          pkgkey: `${name}-${version}`,
+        })}
+      />
+    );
+  }
+
+  let content: JSX.Element;
+
+  if (isLoading) {
+    content = <Loading />;
+  } else if (fetchError) {
+    content = <p>TODO HANDLE ERROR</p>;
+  } else if (assetSavedObjects === undefined) {
+    content = (
+      <EuiTitle>
+        <h2>
+          <FormattedMessage
+            id="xpack.fleet.epm.packageDetails.assets.noAssetsFoundLabel"
+            defaultMessage="No assets found"
+          />
+        </h2>
+      </EuiTitle>
+    );
+  } else {
+    content = (
+      <>
+        {allowedAssetTypes.map((assetType) => {
+          const sectionAssetSavedObjects = assetSavedObjects.filter((so) => so.type === assetType);
+          if (!sectionAssetSavedObjects.length) {
+            return null;
+          }
+
+          return (
+            <>
+              <EuiAccordion
```

Comment:
Maybe part of this component tree is worth moving into a local variable as a `AssetAccordionChild` component or something along those lines? Would clean up the amount of code inside this `map` a bit and allow us to pass props into a component instead on each iteration of the `map`